providers/shared/components/directory/id.ftl

```diff
@@ -0,0 +1,106 @@
+[#ftl]
+
+[@addComponentDeployment
+    type=DIRECTORY_COMPONENT_TYPE
+    defaultGroup="solution"
+/]
+
+[@addComponent
+    type=DIRECTORY_COMPONENT_TYPE
+    properties=
+        [
+            {
+                "Type"  : "Description",
+                "Value" : "A managed directory services instance"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Engine",
+                "Types" : STRING_TYPE,
+                "Values" : [ "Simple", "ActiveDirectory" ],
+                "Mandatory" : true
+            },
+            {
+                "Names" : [ "Certificate", "Hostname" ],
+                "Children" : certificateChildConfiguration
+            },
+            {
+                "Names" : "IPAddressGroups",
+                "Types" : ARRAY_OF_STRING_TYPE,
+                "Default" : []
+            },
+            {
+                "Names" : "ShortName",
+                "Types" : STRING_TYPE
+            },
```

Comment:
Probably good to have a description here to. This is the Netbios name right?

Reply:
Correct. Added